src/main/scala/scorer/Scorer.scala

```diff
@@ -40,12 +40,49 @@ object Scorer {
   def fuseResult(result: WikiListResult): List[String] = {
     val page: WikiListPage = result.page
     val allTypes: Map[String, Int] = result.types
-    val scores: Map[Symbol, Map[String,Double]] = result.scores
+    val scores: Map[Symbol, Map[String, Double]] = result.scores
+
+    if (allTypes.isEmpty || scores.isEmpty) {
+      return List[String]()
+    }
 
     scores
       .map { case (algorithmName, scoredTypes) =>
         val mutatedScores = algorithmName match {
           case 'tfIdf => {
+            // idea: find the biggest gap between 2 scorings and use the lower scoredType as threshold
+            //val scoredTypeCount = scoredTypes.size
+            //var scoreDistanceSum = 0.0 //scoredTypes.foldLeft(0.0) { (acc, scoredType) => acc + scoredType._2 }
+
+            val sortedScoredTypes = scoredTypes
+              .toList
+              .sortBy({ case(_, score) => -score })
+
+            var oldScoredType: (String, Double) = sortedScoredTypes.head
+            var maxScoredType: (String, Double) = sortedScoredTypes.head
+            var maxScoreDistance: Double = 0.0
+            val minScore = sortedScoredTypes.last._2
+
+            sortedScoredTypes.foreach({ case (typeName, score) =>
+              val oldScore = oldScoredType._2
+              //scoreDistanceSum += oldScore - score
+              if (oldScore - score >= maxScoreDistance) {
+                maxScoreDistance = oldScore - score
+                maxScoredType = (typeName, score)
+              }
+
+              oldScoredType = (typeName, score)
+            })
+
+            // val scoreDistanceAvg = scoreDistanceSum / scoredTypeCount
+
+            if (minScore < maxScoreDistance) {
```

Comment:
when do we know the clustering was correct? the drop should be relatively high. I tried comparison with average and fixed values. checking if at least the minimum score is smaller than the largest score drop works quite well for complicated lists. 
most lists with almost no score drop where parsed wrong. i checked about 100 lists.